### ISSUE TITLE
Instance normalization implemented?

### ISSUE THREAD

**Issue by AlonaGolts** [DmitryUlyanov/texture_nets#36]
Hi Dmitry! has the instance normalization layer been implemented already? I haven't found reference in the code. Maybe you can refer me to the file/function that contains this addition. Thanks!

**Comment by DmitryUlyanov** [comment]
There's no explicit name, but it is there in the code.

**Comment by AlonaGolts** [comment]
I'll have another look, thanks for the quick reply!

**Comment by DmitryUlyanov** [comment]
Basically if you perform batchnorm over each object separately you get instance norm.

**Comment by AlonaGolts** [comment]
Hi Dmitry, 

After reading the paper on "instance-norm" more carefully, i now understand what you mean. However, there is a non-symmetry in the training and inference stages of batch-norm, whereas in your paper you set these two stages to act the same. Is that detail also solved by choosing the batch-size to be 1 or is it explicitly implemented in the code? 

Thanks!

**Comment by DmitryUlyanov** [comment]
It is solved. I will update the code with a better implementation shortly.

**Comment by AlonaGolts** [comment]
OK, thanks again!

**Comment by DmitryUlyanov** [comment]
Updated the code with InstanceNorm implementation.
